File: backend/fund-service/schemas/fund.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from decimal import Decimal

class FundBase(BaseModel):
    fund_code: str
    fund_name: str
    fund_type: str
    risk_level: str

class FundResponse(FundBase):
    id: int
    nav: Decimal
    acc_nav: Decimal
    min_purchase: Decimal
    purchase_fee: Decimal
    redeem_fee: Decimal
    manager_name: Optional[str]
    company_name: Optional[str]
    status: str
    created_at: datetime

    class Config:
        from_attributes = True

class FundNavHistoryResponse(BaseModel):
    fund_code: str
    nav_date: datetime
    nav: Decimal
    acc_nav: Decimal

File: frontend/fund_app/lib/features/fund/domain/entities/fund.dart
import 'package:equatable/equatable.dart';

/// Fund Entity - 基金实体
class Fund extends Equatable {
  final int id;
  final String fundCode;
  final String fundName;
  final String fundType;
  final String riskLevel;
  final String nav;
  final String accNav;
  final String minPurchase;
  final String purchaseFee;
  final String redeemFee;
  final String? managerName;
  final String? companyName;
  final String? description;
  final String status;
  final DateTime? createdAt;

  const Fund({
    required this.id,
    required this.fundCode,
    required this.fundName,
    required this.fundType,
    required this.riskLevel,
    required this.nav,
    required this.accNav,
    required this.minPurchase,
    required this.purchaseFee,
    required this.redeemFee,
    this.managerName,
    this.companyName,
    this.description,
    required this.status,
    this.createdAt,
  });

  String get fundTypeName {
    switch (fundType) {
      case 'money_market': return '货币基金';
      case 'stock': return '股票基金';
      case 'hybrid': return '混合基金';
      case 'bond': return '债券基金';
      case 'fof': return 'FOF基金';
      default: return fundType;
    }
  }

  @override
  List<Object?> get props => [id, fundCode, fundName, fundType, riskLevel];
}
